src/main/java/com/infomaniak/lib/core/bugtracker/BugTrackerActivity.kt

```diff
@@ -0,0 +1,260 @@
+import android.database.Cursor
+import android.net.Uri
+import android.os.Build
+import android.os.Bundle
+import android.provider.OpenableColumns
+import android.text.format.Formatter
+import android.webkit.MimeTypeMap
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.activity.viewModels
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.database.getStringOrNull
+import androidx.core.view.isGone
+import androidx.core.view.isVisible
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.lifecycleScope
+import androidx.navigation.navArgs
+import com.infomaniak.lib.core.R
+import com.infomaniak.lib.core.databinding.ActivityBugTrackerBinding
+import com.infomaniak.lib.core.networking.HttpClient.okHttpClient
+import com.infomaniak.lib.core.networking.HttpUtils.getHeaders
+import com.infomaniak.lib.core.utils.SnackbarUtils.showSnackbar
+import com.infomaniak.lib.core.utils.whenResultIsOk
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import okhttp3.MediaType.Companion.toMediaTypeOrNull
+import okhttp3.MultipartBody
+import okhttp3.Request
+import okhttp3.RequestBody.Companion.toRequestBody
+
+class BugTrackerActivity : AppCompatActivity() {
+
+    private val binding: ActivityBugTrackerBinding by lazy { ActivityBugTrackerBinding.inflate(layoutInflater) }
+    private val bugTrackerViewModel: BugTrackerViewModel by viewModels()
+    private val navigationArgs: BugTrackerActivityArgs by navArgs()
+
+    private val imageAdapter = BugTrackerImageAdapter() { updateImageTotalSize() }
+    val types = listOf("bugs", "features")
+    var type = types[DEFAULT_REPORT_TYPE]
+
+    private val selectImagesResultLauncher = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) {
+        it.whenResultIsOk { data -> onImagesImported(data) }
+    }
+
+    private fun onImagesImported(importIntent: Intent?) {
+        val clipData = importIntent?.clipData
+        val uri = importIntent?.data
+        var errorCount = 0
+
+        val newImages = mutableListOf<BugTrackerImage>()
+
+        try {
+            if (clipData != null) {
+                for (i in 0 until clipData.itemCount) {
+                    runCatching {
+                        getImageFromUri(clipData.getItemAt(i).uri)?.let(newImages::add)
+                    }.onFailure {
+                        it.printStackTrace()
+                        errorCount++
+                    }
+                }
+            } else if (uri != null) {
+                getImageFromUri(uri)?.let(newImages::add)
+            }
+        } catch (exception: Exception) {
+            exception.printStackTrace()
+            errorCount++
+        } finally {
+            if (errorCount > 0) {
+                showSnackbar("Some images failed to be imported: ($errorCount)")
+            }
+        }
+
+        imageAdapter.addImages(newImages)
+        updateImageTotalSize()
+    }
```

Comment:
Factor with reference to kDrive and put this method in Core, as it will also be used for kMail
https://github.com/Infomaniak/android-kDrive/blob/master/app/src/main/java/com/infomaniak/drive/ui/addFiles/ImportFilesDialog.kt#L86